pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -141,14 +151,10 @@ func (in *NodeClaimSpec) convertFrom(ctx context.Context, v1beta1nc *v1beta1.Nod
 	})
 
 	defaultNodeClassGVK := injection.GetNodeClasses(ctx)[0]
-	nodeclassGroupVersion, err := schema.ParseGroupVersion(v1beta1nc.NodeClassRef.APIVersion)
-	if err != nil {
-		return "", err
-	}
 	in.NodeClassRef = &NodeClassReference{
 		Name:  v1beta1nc.NodeClassRef.Name,
 		Kind:  lo.Ternary(v1beta1nc.NodeClassRef.Kind == "", defaultNodeClassGVK.Kind, v1beta1nc.NodeClassRef.Kind),
-		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, lo.Ternary(nodeclassGroupVersion.Group == "", nodeclassGroupVersion.Version, nodeclassGroupVersion.Group)),
+		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, strings.Split(v1beta1nc.NodeClassRef.APIVersion, "/")[0]),
```

Comment:
I think I had the same question for Jigisha in her PR, why would we remove the upstream function for this and implement it ourselves? Seems like it's working as expected as is, no?

Reply:
You did and I think the upstream function is more confusing than this is. The string split is the most straightforward way to do this IMO. I had a lot of questions looking at the current implementation, wondering why we are setting it to version in some cases and not others

Reply:
Does our CEL validation cover the cases that the function covers? e.g. two `/` is invalid

Reply:
It's a good callout. We could/should probably add validation for this